apps/web/src/components/Composer/NewPublication.tsx

```diff
@@ -241,7 +236,7 @@ const NewPublication: FC<NewPublicationProps> = ({ publication }) => {
 
     // Track in leafwatch
     const eventProperties = {
-      comment_on: isComment ? publication.id : null,
+      comment_on: isComment ? publication?.id : null,
```

Comment:
`<NewPublication>` accepts an optional `publication` prop. See the following two usages:

- https://github.com/ocavue/hey/blob/99f9c521dc993b216b443605ffdf02a1e75094b1/apps/web/src/components/Publication/index.tsx#L158

- https://github.com/ocavue/hey/blob/99f9c521dc993b216b443605ffdf02a1e75094b1/apps/web/src/components/Shared/GlobalModals.tsx#L133


Thus, the type of `publication` should be `publication?: MirrorablePublication`. TypeScript didn't complain about this issue before my PR because the previous `withLexicalContext` HOC just marks the component props [as `any`](https://github.com/ocavue/hey/blob/99f9c521dc993b216b443605ffdf02a1e75094b1/apps/web/src/components/Shared/Lexical/withLexicalContext.tsx#L27). I removed `withLexicalContext` and also correct this type error.